docs: add doc for assertions & data contracts (#10029)
datahub-project/datahub: docs/managed-datahub/observe/data-contract.md

# Data Contracts

## What Is a Data Contract

> The definition of a Data Contract is consisted of the following:

Reviewer: we probably should have a high level sentence about what a data contract is, before diving into the properties

</details>

<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/data_contracts/validated-data-contracts-ui.png"/>
</p>

## Data Contract and Assertions

Another way to word our vision of data contracts is **A bundle of verifiable assertions on physical data assets representing a public producer commitment.** 
These can be all the assertions on an asset or only the subset you want publicly promised to consumers. Data Contracts allow you to **promote a selected group of your assertions** as a public promise: if this subset of assertions is not met, the Data Contract is failing.

See docs on [assertions](/docs/managed-datahub/observe/assertions.md) for more details on the types of assertions and how to create and run them.

:::note Ownership 

> the owner of the physical data asset is also the owner of the contract and can accept proposed changes and make changes themselves to the contract.

Reviewer: capitalization

:::

## How to Create Data Contracts

Data Contracts can be created via DataHub CLI (YAML), API, or UI.

### DataHub CLI using YAML

For creation via CLI, it’s a simple CLI upsert command that you can integrate into your CI/CD system to publish your Data Contracts and any change to them.

1. Define your data contract.

> ```yaml

Reviewer: ideally we should inline this from a separate file

#                     policyId:
#                       type: integer
#                     fileId:
#                       type: integer
#     required:
#       - field_bar
#       - field_documents

```
2. Use the CLI to create the contract by running the below command.

```shell
datahub datacontract upsert -f contract_definition.yml
```

3. Now you can see your contract on the UI.

<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/data_contracts/data-contracts-ui.png"/>
</p>

### UI

1. Navigate to the Dataset Profile for the dataset you wish to create a contract for
2. Under the **Validations** > **Data Contracts** tab, click **Create**.

<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/data_contracts/create-data-contract-ui.png"/>
</p>

3. Select the assertions you wish to be included in the Data Contract.

<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/data_contracts/select-assertions.png"/>
</p>

:::note Create Data Contracts via UI

> Please note that when creating a Data Contract via UI, the Freshness, Schema, and Data Quality assertions are expected to have been created already.

Reviewer: ```suggestion
When creating a Data Contract via UI, the Freshness, Schema, and Data Quality assertions must be created first.
```

Reviewer: imo the note section is a bit large and interrupts the flow here

:::
4. Now you can see it in the UI.

<p align="center">
  <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/observe/data_contracts/contracts-created.png"/>
</p>

### API

> :::note

Reviewer: don't need a `note` here
